lib/cisco_node_utils/bgp.rb

```diff
@@ -327,10 +327,19 @@ def enforce_first_as
       config_get('bgp', 'enforce_first_as', @get_args)
     end
 
-    def enforce_first_as=(state)
```

Comment:
@glennmatthews Reworked both log neighbor changes and enforce first as - as that was wrong too.

Rubocop happy.

Minitest happy.

Tested puppet - looks good:

Both set to true(default).

RP/0/RP0/CPU0:ios#sh run router bgp
Wed Nov 18 19:38:52.558 UTC
router bgp 55
 bgp confederation identifier 33
 bgp router-id 1.1.1.1
 bgp cluster-id 55
 bgp graceful-restart restart-time 131
 bgp graceful-restart stalepath-time 311
 bgp graceful-restart
 bgp bestpath med confed
 vrf blue
  rd auto
  timers bgp 46 111
  bgp router-id 192.168.0.66
  bgp bestpath cost-community ignore
  bgp bestpath med always
  bgp bestpath compare-routerid
  bgp bestpath as-path multipath-relax
 !
!

Both set to false (disable) in the global VRF

RP/0/RP0/CPU0:ios#sh run router bgp
Wed Nov 18 19:39:49.768 UTC
router bgp 55
 bgp confederation identifier 33
 bgp router-id 1.1.1.1
 bgp cluster-id 55
 bgp graceful-restart restart-time 131
 bgp log neighbor changes disable
 bgp graceful-restart stalepath-time 311
 bgp graceful-restart
 bgp bestpath med confed
 bgp enforce-first-as disable
 vrf blue
  rd auto
  timers bgp 46 111
  bgp router-id 192.168.0.66
  bgp bestpath cost-community ignore
  bgp bestpath med always
  bgp bestpath compare-routerid
  bgp bestpath as-path multipath-relax
 !
!

Both set to false (disable) in the VRF

sh run router bgp
Wed Nov 18 19:40:57.464 UTC
router bgp 55
 bgp confederation identifier 33
 bgp router-id 1.1.1.1
 bgp cluster-id 55
 bgp graceful-restart restart-time 131
 bgp log neighbor changes disable
 bgp graceful-restart stalepath-time 311
 bgp graceful-restart
 bgp bestpath med confed
 bgp enforce-first-as disable
 vrf blue
  rd auto
  timers bgp 46 111
  bgp router-id 192.168.0.66
  bgp log neighbor changes disable
  bgp bestpath cost-community ignore
  bgp bestpath med always
  bgp bestpath compare-routerid
  bgp bestpath as-path multipath-relax
  bgp enforce-first-as disable
 !
!